calico_kubernetes/calico_kubernetes.py

```diff
@@ -141,6 +137,9 @@ def status(self, namespace, pod_name, docker_id):
                 workload_id=self.docker_id
             )
         except KeyError:
+            # Obtain information from Docker Client and validate container state
+            # Will exit safely (0) in host NetworkMode
+            self._validate_container_state(self.docker_id)
```

Comment:
I don't think this is the right abstraction.  We may or may not care about all of the things that currently, or might be one day be done in `_validate_container_state`.  We do know that we can't report a status for host-networked pods.

It is also a bit confusing to rely on the sys.exit in _validate_container_state here - makes the code harder to read.

I had actually already worked up a fix for this issue in another PR (which may or may not be the right way) - take a look here: https://github.com/projectcalico/calico-kubernetes/pull/89/files#diff-869f81c0955f1bd9cf59f97823a22585R142.

Let me know what you think, and we can discuss the right fix.